ci-operator/step-registry/upi/gcp/ssh-bastion/pre/upi-gcp-ssh-bastion-pre-commands.sh

```diff
@@ -0,0 +1,278 @@
+}
+print_logs &
+squid -N -f /squid/squid.conf
+EOF
+)"
+
+# create ignition entries for certs and script to start squid and systemd unit entry
+# create the proxy instance and then get its IP
+
+generate_proxy_ignition
+
+# we need to be able to tear down the proxy even if install fails
+# cannot rely on presence of ${SHARED_DIR}/metadata.json
+echo "${REGION}" >> "${SHARED_DIR}/proxyregion"
+
+gcloud compute instances create "${CLUSTER_NAME}-bastion" \
+  --image=${IMAGE_NAME} \
+  --image-project=${IMAGE_PROJECT} \
+  --metadata-from-file=user-data=/tmp/proxy.ign \
+  --machine-type=${MACHINE_TYPE} \
+  --network=${NETWORK} \
+  --subnet=${CONTROL_PLANE_SUBNET} \
+  --zone=${ZONE_0} \
+  --tags="${CLUSTER_NAME}-bastion"
+
+echo "Created bastion instance"
+echo "Waiting for the proxy service starting running..." && sleep 60s
+
+if [[ -s "${SHARED_DIR}/xpn.json" ]]; then
+  HOST_PROJECT="$(jq -r '.hostProject' "${SHARED_DIR}/xpn.json")"
+  project_option="--project=${HOST_PROJECT}"
+else
+  project_option=""
+fi
+gcloud ${project_option} compute firewall-rules create "${CLUSTER_NAME}-bastion-ingress-allow" \
+  --network ${NETWORK} \
+  --allow tcp:22,tcp:3128,tcp:3129,tcp:5000,tcp:8080 \
+  --target-tags="${CLUSTER_NAME}-bastion"
+cat > "${SHARED_DIR}/ssh-bastion-destroy.sh" << EOF
+gcloud compute instances delete -q "${CLUSTER_NAME}-bastion" --zone=${ZONE_0}
+gcloud ${project_option} compute firewall-rules delete -q "${CLUSTER_NAME}-bastion-ingress-allow"
+EOF
+
+if [ X"${DISCONNECTED_NETWORK}" == X"yes" ]; then
+  gcloud ${project_option} compute firewall-rules create "${CLUSTER_NAME}-bastion-egress-allow" --allow='all' --direction=EGRESS --network="${NETWORK}" --target-tags="${CLUSTER_NAME}-bastion"
+  cat >> "${SHARED_DIR}/ssh-bastion-destroy.sh" << EOF
+gcloud ${project_option} compute firewall-rules delete -q "${CLUSTER_NAME}-bastion-egress-allow"
+EOF
+fi
+
+INSTANCE_ID="${CLUSTER_NAME}-bastion"
+echo "Instance ${INSTANCE_ID}"
+
+# to allow log collection during gather:
+# append to proxy instance ID to "${SHARED_DIR}/gcp-instance-ids.txt"
+echo "${INSTANCE_ID}" >> "${SHARED_DIR}/gcp-instance-ids.txt"
+
+gcloud compute instances list --filter="name=('${INSTANCE_ID}')" \
+  --zones "${ZONE_0}" --format json > /tmp/${INSTANCE_ID}-bastion.json
+PRIVATE_PROXY_IP="$(jq -r '.[].networkInterfaces[0].networkIP' /tmp/${INSTANCE_ID}-bastion.json)"
+PUBLIC_PROXY_IP="$(jq -r '.[].networkInterfaces[0].accessConfigs[0].natIP' /tmp/${INSTANCE_ID}-bastion.json)"
+
+# echo proxy IP to ${SHARED_DIR}/proxyip
+echo "${PUBLIC_PROXY_IP}" >> "${SHARED_DIR}/proxyip"
+
+if [ X"${DISCONNECTED_NETWORK}" == X"yes" ]; then
+  PROXY_URL="http://${CLUSTER_NAME}:${PASSWORD}@${PRIVATE_PROXY_IP}:3128/"
+  # due to https://bugzilla.redhat.com/show_bug.cgi?id=1750650 we don't use a tls end point for squid
+
+  cat >> "${CONFIG}" << EOF
+proxy:
+  httpsProxy: ${PROXY_URL}
+  httpProxy: ${PROXY_URL}
+EOF
+fi
+
+if [ X"${PUBLISH_STRATEGY}" == X"Internal" ]; then
```

Comment:
Per my understanding, this script is used to create bastion host, the change to install-config.yaml is better to be moved out of this script.

Reply:
Similar reason to the last one.